src/Domain.LinnApps/ImportBooks/ImportBookService.cs

```diff
@@ -3,19 +3,84 @@
     using System;
     using System.Collections.Generic;
     using System.Linq;
+
     using Linn.Common.Persistence;
+    using Linn.Stores.Domain.LinnApps.Exceptions;
+    using Linn.Stores.Domain.LinnApps.ExternalServices;
+    using Linn.Stores.Domain.LinnApps.Models;
 
     public class ImportBookService : IImportBookService
     {
         private readonly IRepository<ImportBookExchangeRate, ImportBookExchangeRateKey> exchangeRateRepository;
+
         private readonly IRepository<LedgerPeriod, int> ledgerPeriodRepository;
 
+        private readonly IRepository<ImportBookOrderDetail, ImportBookOrderDetailKey> orderDetailRepository;
+
+        private readonly IRepository<PurchaseLedger, int> purchaseLedgerRepository;
+
+
+        private readonly IPurchaseLedgerPack purchaseLedgerPack;
+
+        private readonly IQueryRepository<Supplier> supplierRepository;
+
+        private readonly ITransactionManager transactionManager;
+
         public ImportBookService(
             IRepository<ImportBookExchangeRate, ImportBookExchangeRateKey> exchangeRateRepository,
-            IRepository<LedgerPeriod, int> ledgerPeriodRepository)
+            IRepository<LedgerPeriod, int> ledgerPeriodRepository,
+            IQueryRepository<Supplier> supplierRepository,
+            IRepository<ImportBookOrderDetail, ImportBookOrderDetailKey> orderDetailRepository,
+            IRepository<PurchaseLedger, int> purchaseLedgerRepository,
+            ITransactionManager transactionManager,
+            IPurchaseLedgerPack purchaseLedgerPack)
         {
             this.exchangeRateRepository = exchangeRateRepository;
             this.ledgerPeriodRepository = ledgerPeriodRepository;
+            this.supplierRepository = supplierRepository;
+            this.orderDetailRepository = orderDetailRepository;
+            this.transactionManager = transactionManager;
+            this.purchaseLedgerPack = purchaseLedgerPack;
+            this.purchaseLedgerRepository = purchaseLedgerRepository;
+        }
+
+        public IEnumerable<ImportBookExchangeRate> GetExchangeRates(string date)
+        {
+            var unformattedDate = DateTime.Parse(date);
+            var formattedDate = unformattedDate.ToString("MMMYYYY");
+
+            var ledgerPeriod = this.ledgerPeriodRepository.FindBy(x => x.MonthName == formattedDate);
+
+            var exchangeRates = this.exchangeRateRepository.FilterBy(x => x.PeriodNumber == ledgerPeriod.PeriodNumber);
+
+            return exchangeRates;
+        }
+
+        public ProcessResult PostDutyForOrderDetails(
+            IEnumerable<ImportBookOrderDetail> orderDetails,
+            int supplierId,
+            int employeeId,
+            DateTime postDutyDate)
+        {
+            foreach (var detail in orderDetails)
+            {
+                var oldDetail = this.orderDetailRepository.FindById(
```

Comment:
do any order details have to be passed in here? 
Shouldn't it just deal with what's in the database and disable the button while there are any outstanding changes on the form?